### PR TITLE
CB-420 Rename Blueprint to Cluster definition in the background

### PR DESCRIPTION
**Pull request by mhmxs**

### CODE REVIEW

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceEvent.java [hunk 1]
```diff
@@ -3,8 +3,8 @@
 public enum ResourceEvent {
     WORKSPACE_CREATED("resource.workspace.created"),
     WORKSPACE_DELETED("resource.workspace.deleted"),
-    BLUEPRINT_CREATED("resource.blueprint.created"),
-    BLUEPRINT_DELETED("resource.blueprint.deleted"),
+    BLUEPRINT_CREATED("resource.clusterdefinition.created"),
```
Comment (keyki) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceEvent.java]: Should we rename the enums as well?

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/ClusterDefinition.java [hunk 2]
```diff
@@ -33,12 +33,13 @@
     @Column(nullable = false)
     private String name;
 
-    @Column(nullable = false)
+    @Column(name = "blueprintText", nullable = false)
```
Comment (doktoric) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/ClusterDefinition.java]: can we rename this ?

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/cluster/Cluster.java [hunk 3]
```diff
@@ -138,8 +140,8 @@
 
     @Convert(converter = SecretToString.class)
     @SecretValue
-    @Column(nullable = false)
-    private Secret extendedBlueprintText = Secret.EMPTY;
+    @Column(name = "extendedBlueprintText", nullable = false)
```
Comment (doktoric) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/cluster/Cluster.java]: can we rename the column ?

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceEvent.java (hunk 1 above)
Comment (mhmxs) [reply cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ResourceEvent.java]: There is a separated issue to rename public part

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintController.java [hunk 4]
```diff
@@ -33,14 +33,14 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(BlueprintController.class);
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintController.java]: Should we rename this class as well?

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/cluster/Cluster.java (hunk 3 above)
Comment (mhmxs) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/cluster/Cluster.java]: I would like to do the migration at the last step to keep backward compatible as late as possible

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintV3Controller.java [hunk 5]
```diff
@@ -20,22 +20,22 @@
...
+import com.sequenceiq.cloudbreak.service.blueprint.ClusterDefinitionService;
 import com.sequenceiq.cloudbreak.service.user.UserService;
 import com.sequenceiq.cloudbreak.service.workspace.WorkspaceService;
 import com.sequenceiq.cloudbreak.util.WorkspaceEntityType;
 
 @Controller
 @Transactional(TxType.NEVER)
-@WorkspaceEntityType(Blueprint.class)
+@WorkspaceEntityType(ClusterDefinition.class)
 public class BlueprintV3Controller extends NotificationController implements BlueprintV3Endpoint {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintV3Controller.java]: Should we rename this class as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintController.java (hunk 4 above)
Comment (mhmxs) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintController.java]: This issue is not about the public facing of Blueprints, we have separated issue to rename the public part

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/BlueprintValidationExceptionMapper.java [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
 
-import com.sequenceiq.cloudbreak.blueprint.validation.BlueprintValidationException;
+import com.sequenceiq.cloudbreak.template.validation.ClusterDefinitionValidationException;
 import org.springframework.stereotype.Component;
 
 import javax.ws.rs.core.Response.Status;
 
 @Component
-public class BlueprintValidationExceptionMapper extends BaseExceptionMapper<BlueprintValidationException> {
+public class BlueprintValidationExceptionMapper extends BaseExceptionMapper<ClusterDefinitionValidationException> {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/mapper/BlueprintValidationExceptionMapper.java]: Should we rename this class as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintV3Controller.java (hunk 5 above)
Comment (mhmxs) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/BlueprintV3Controller.java]: Already answered

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java [hunk 7]
```diff
@@ -15,73 +15,73 @@
...
+import com.sequenceiq.cloudbreak.domain.ClusterDefinition;
 import com.sequenceiq.cloudbreak.domain.json.Json;
 import com.sequenceiq.cloudbreak.json.CloudbreakApiException;
 import com.sequenceiq.cloudbreak.json.JsonHelper;
 import com.sequenceiq.cloudbreak.service.MissingResourceNameGenerator;
 import com.sequenceiq.cloudbreak.util.JsonUtil;
 
 @Component
-public class BlueprintRequestToBlueprintConverter extends AbstractConversionServiceAwareConverter<BlueprintRequest, Blueprint> {
+public class BlueprintRequestToBlueprintConverter extends AbstractConversionServiceAwareConverter<BlueprintRequest, ClusterDefinition> {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java]: Should we rename this class as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintToBlueprintResponseConverter.java [hunk 8]
```diff
@@ -5,23 +5,23 @@
 import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.api.model.BlueprintResponse;
-import com.sequenceiq.cloudbreak.domain.Blueprint;
+import com.sequenceiq.cloudbreak.domain.ClusterDefinition;
 
 @Component
-public class BlueprintToBlueprintResponseConverter extends AbstractConversionServiceAwareConverter<Blueprint, BlueprintResponse> {
+public class BlueprintToBlueprintResponseConverter extends AbstractConversionServiceAwareConverter<ClusterDefinition, BlueprintResponse> {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintToBlueprintResponseConverter.java]: Should we rename this class as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintViewToBlueprintViewResponseConverter.java [hunk 9]
```diff
@@ -3,12 +3,12 @@
 import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.api.model.BlueprintViewResponse;
-import com.sequenceiq.cloudbreak.domain.view.BlueprintView;
+import com.sequenceiq.cloudbreak.domain.view.ClusterDefinitionView;
 
 @Component
-public class BlueprintViewToBlueprintViewResponseConverter extends CompactViewToCompactViewResponseConverter<BlueprintView, BlueprintViewResponse> {
+public class BlueprintViewToBlueprintViewResponseConverter extends CompactViewToCompactViewResponseConverter<ClusterDefinitionView, BlueprintViewResponse> {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintViewToBlueprintViewResponseConverter.java]: Should we rename this class as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/v2/cli/BlueprintToBlueprintRequestConverter.java [hunk 10]
```diff
@@ -6,20 +6,20 @@
 
 import com.sequenceiq.cloudbreak.api.model.BlueprintRequest;
 import com.sequenceiq.cloudbreak.converter.AbstractConversionServiceAwareConverter;
-import com.sequenceiq.cloudbreak.domain.Blueprint;
+import com.sequenceiq.cloudbreak.domain.ClusterDefinition;
 
 @Component
 public class BlueprintToBlueprintRequestConverter
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/v2/cli/BlueprintToBlueprintRequestConverter.java]: Should we rename this class as well?

File: core/src/main/java/com/sequenceiq/cloudbreak/init/clusterdefinition/DefaultAmbariBlueprintCache.java [hunk 11]
```diff
@@ -19,17 +19,17 @@
...
-import com.sequenceiq.cloudbreak.blueprint.utils.BlueprintUtils;
+import com.sequenceiq.cloudbreak.clusterdefinition.utils.AmbariBlueprintUtils;
 import com.sequenceiq.cloudbreak.converter.BlueprintRequestToBlueprintConverter;
-import com.sequenceiq.cloudbreak.domain.Blueprint;
+import com.sequenceiq.cloudbreak.domain.ClusterDefinition;
 import com.sequenceiq.cloudbreak.domain.json.Json;
 
 @Service
-public class DefaultBlueprintCache {
+public class DefaultAmbariBlueprintCache {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/init/clusterdefinition/DefaultAmbariBlueprintCache.java]: Should we rename this class as well?

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/ClusterDefinition.java (hunk 2 above)
Comment (mhmxs) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/ClusterDefinition.java]: I would like to do the migration at the last step to keep backward compatible as late as possible

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/AmbariBlueprintPortConfigCollector.java [hunk 12]
```diff
@@ -12,26 +12,26 @@
...
 import com.sequenceiq.cloudbreak.api.model.ExposedService;
-import com.sequenceiq.cloudbreak.domain.Blueprint;
-import com.sequenceiq.cloudbreak.template.processor.BlueprintTextProcessor;
+import com.sequenceiq.cloudbreak.domain.ClusterDefinition;
+import com.sequenceiq.cloudbreak.template.processor.AmbariBlueprintTextProcessor;
 
 @Component
 @ConfigurationProperties
-public class BlueprintPortConfigCollector {
+public class AmbariBlueprintPortConfigCollector {
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/AmbariBlueprintPortConfigCollector.java]: Should we rename this class as well?

**Comment by keyki** [comment]
I think we should revert this a rebase onto the api-refatctor branch.

**Pull request merged**